Sources/TUSKit/TUSClient.swift

```diff
@@ -16,7 +16,7 @@ public protocol TUSClientDelegate: AnyObject {
     /// TUSClient is starting an upload
     func didStartUpload(id: UUID, context: [String: String]?, client: TUSClient)
     /// `TUSClient` just finished an upload, returns the URL of the uploaded file.
-    func didFinishUpload(id: UUID, url: URL, context: [String: String]?, client: TUSClient)
+    func didFinishUpload(id: UUID, url: URL, responseHeaders: [String: String]?, context: [String: String]?, client: TUSClient)
```

Comment:
Adding support for response headers this way makes this PR a breaking change which I want to avoid.

We could have both versions of `didFinishUpload` with and without the `responseHeaders` defined on this protocol. Both would need to be called from the places where we call `delegate?.didFinishUpload` which would be error prone so expanding the test suite to verify we always call both would be desirable.

If you have other ideas to implement this as a non-breaking change, I'm all ears 👍🏼

Reply:
I've made the above change making `func didFinishUpload(id: UUID, url: URL, context: [String: String]?, client: TUSClient, responseHeaders: [String: String]?)` an optional function and called both the functions in a private function.
In the mock test, I've just added a skeleton that confirms to the optional delegate method. Could you please suggest me to how to make the test case work with the changes?

Reply:
Sure. Have a nice vacation.
Will be looking forward for your feedback after your vacation.